ipv8/util.py

```diff
@@ -38,3 +39,17 @@ async def call_async(*args, **kwargs):
 
 def strip_sha1_padding(s: bytes):
     return s[12:] if s.startswith(b'SHA-1\x00\x00\x00\x00\x00\x00\x00') else s
+
+
+def create_event_with_signals(*args):
+    event = Event()
+
+    for sig in (args or (signal.SIGINT, signal.SIGTERM)):
+        signal.signal(sig, lambda _, __: event.set())
+
+    return event
+
+
+def run_forever(return_event=False):
+    # Helper function for waiting until the user presses Ctrl+C.
+    return create_event_with_signals().wait()
```

Comment:
Please add docstrings and types to these functions.

Reply:
Will do.